config/src/main/scala/bloop/config/util/ConfigUtil.scala

```diff
@@ -1,26 +1,14 @@
 package bloop.config.util
 
-import java.nio.file.{Path, Files}
+import java.nio.file.Path
 
 object ConfigUtil {
   def pathsOutsideRoots(roots: Seq[Path], paths: Seq[Path]): Seq[Path] = {
-    paths.filterNot { path =>
-      roots.exists { root =>
-        var found: Boolean = false
-        val rootDirSize = root.toString.size
-        var currentTarget = (if (Files.isRegularFile(path)) path.getParent else path).toAbsolutePath
-        while (!found &&
-               currentTarget != null &&
-               // Use a heuristic to know if we should short-circuit and return false
-               currentTarget.toString.size >= rootDirSize) {
-          if (currentTarget == root) {
-            found = true
-          }
-
-          currentTarget = currentTarget.getParent
-        }
-        found
-      }
+    for {
```

Comment:
this change doesn't necessarily have to be here, but this implementation is way more straightforward :)